### PR TITLE
Avoids running proxy commands if ssh_proxy config is empty/blank

### LINKED ISSUE #2267: Execution of ssh_proxy if block when the configuration variable is empty
This is similar to #1936, but slightly different.

Take the following config extract:
```
source:
  default: sql
  sql:
    adapter: mysql2
...
    map:
      name: hostname
      ip: name
      model: model
      group: group
      username: username
      password: password
    vars_map:
      ssh_proxy: ssh_proxy
      enable: enable
```
      
and the following mysql table row:
```
hostid	|hostname	|name		|model		|group		|username	|password	|ssh_proxy	|enable	|comware_cmdline
11505	|myhost		|192.168.20.1	|netscaler	|<GROUPNAME>	|<USER>		|<redacted>	|		|	|512900
```

Currently, even though I've not set a value for `ssh_proxy`, the code block in this [if statement](https://github.com/ytti/oxidized/blob/c0dae48928ec391c01070a32c2eedb034d983c10/lib/oxidized/input/ssh.rb#L137) is executed, presumably because `""` in Ruby considered `true`:

```
Feb 23 12:29:14 nms oxidized[146462]: 192.168.20.1 raised Net::SSH::Proxy::ConnectError (rescued RuntimeError) with msg "command failed: ssh -o StrictHostKeyChecking=no  -W 192.168.20.1:22"
```

The linked if statement should be modified to grab the variable and check that it's not `nil` or `empty` before executing the corresponding code block.

### PR DESCRIPTION
**Pull request by jameskirsop**
## Pre-Request Checklist

- [ ] Passes rubocop code analysis (try `rubocop --auto-correct`)
- [ ] Tests added or adapted (try `rake test`)
- [x] Changes are reflected in the documentation
- [x] User-visible changes appended to [CHANGELOG.md](/CHANGELOG.md)

## Description
This prevents the [ssh_proxy code block](https://github.com/ytti/oxidized/blob/c0dae48928ec391c01070a32c2eedb034d983c10/lib/oxidized/input/ssh.rb#L137) being executed when the value is empty in an SQL source, and otherwise has no effect.

It brings Oxidized in line with the _expected_ behaviour and closes #2267

### CODE REVIEW

**Comment by mortzu** [comment]
Can you please update the changelog? I will merge then

**Comment by mortzu** [comment]
Please process the checklist

**Comment by jameskirsop** [comment]
@mortzu, changelog has been updated.

File: lib/oxidized/input/ssh.rb [hunk 1]
```diff
@@ -134,7 +134,8 @@ def make_ssh_opts
       ssh_opts[:auth_methods] = auth_methods
       Oxidized.logger.debug "AUTH METHODS::#{auth_methods}"
 
-      if (proxy_host = vars(:ssh_proxy))
+      proxy_host = vars(:ssh_proxy)
+      if ( defined?(proxy_host) && !proxy_host.nil? && !proxy_host.empty? )
```
Comment (mjbnz) [inline lib/oxidized/input/ssh.rb:138]: rubocop complains about the parenthesis here, I've removed them in PR #2476

**Pull request merged**